File: 2023-06-20-08-28-07/problems/find_the_highest_altitude/solution.c
int largestAltitude(int* gain, int gainSize){
    int* prefixSum = (int*)malloc((gainSize + 1) * sizeof(int));
    *prefixSum = 0;
    int maxAltitude = 0;

    for (int i = 1; i <= gainSize; i++) {
        *(prefixSum +i) = *(prefixSum + (i - 1)) + *(gain + (i - 1));
        if (*(prefixSum + i) > maxAltitude) {
            maxAltitude = *(prefixSum + i);
        }
    }

    free(prefixSum);  

    return maxAltitude;
}

File: 2023-06-20-08-28-07/problems/find_the_highest_altitude/solution.cs
public class Solution {
    public int LargestAltitude(int[] gain) {
        int n = gain.Length;
        int[] prefixSum = new int[n + 1];
        int maxAltitude = 0;

        for (int i = 1; i <= n; i++)
        {
            prefixSum[i] = prefixSum[i - 1] + gain[i - 1];
            if (prefixSum[i] > maxAltitude)
            {
                maxAltitude = prefixSum[i];
            }
        }

        return maxAltitude;
    }
}